lib/levanter/src/levanter/layers/attention.py

```diff
@@ -764,6 +753,47 @@ def _maybe_flatten(q, axes, name):
     return q
 
 
+def _prepare_sinks_for_splash(attn_sink: NamedArray, q_class, physical_axes_q: PartitionSpec):
+    """Reshape and broadcast attention sinks to (B, H) for the splash kernel."""
+
+    batch_axes = tuple(q_class["B"])
+    head_axes = tuple(q_class["H"])
+    allowed_axes = {ax.name for ax in batch_axes + head_axes}
+    extra_axes = tuple(ax for ax in attn_sink.axes if ax.name not in allowed_axes)
+    if extra_axes:
+        raise NotImplementedError(
+            f"Attention sinks contain axes unsupported by splash attention: {', '.join(ax.name for ax in extra_axes)}"
+        )
+
+    sink = attn_sink
+    sink_axis_names = {ax.name for ax in sink.axes}
+    for ax in batch_axes:
+        if ax.name not in sink_axis_names:
+            sink = sink.broadcast_axis(ax)
+            sink_axis_names.add(ax.name)
+    for ax in head_axes:
+        if ax.name not in sink_axis_names:
+            sink = sink.broadcast_axis(ax)
+            sink_axis_names.add(ax.name)
+
+    if batch_axes:
+        sink = sink.flatten_axes(tuple(ax.name for ax in batch_axes), "splash_batch")
+    else:
+        sink = sink.broadcast_axis(Axis("splash_batch", 1))
+
+    if head_axes:
+        sink = sink.flatten_axes(tuple(ax.name for ax in head_axes), "splash_head")
+    else:
+        sink = sink.broadcast_axis(Axis("splash_head", 1))
+
+    sink = sink.rearrange(("splash_batch", "splash_head"))
```

Comment:
We should probably move _maybe_flayten outside of the bshd function and use it here.